internal/provider/resource_service_account.go

```diff
@@ -0,0 +1,337 @@
+package provider
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	tharsis "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg"
+	ttypes "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg/types"
+)
+
+// OIDCTrustPolicyModel is the model for a trust policy.
+type OIDCTrustPolicyModel struct {
+	BoundClaims map[string]types.String `tfsdk:"bound_claims"`
+	Issuer      types.String            `tfsdk:"issuer"`
+}
+
+// ServiceAccountModel is the model for a service account.
+// Fields intentionally omitted: NamespaceMemberships and ActivityEvents.
+type ServiceAccountModel struct {
+	ID                types.String           `tfsdk:"id"`
+	ResourcePath      types.String           `tfsdk:"resource_path"`
+	Name              types.String           `tfsdk:"name"`
+	Description       types.String           `tfsdk:"description"`
+	GroupPath         types.String           `tfsdk:"group_path"`
+	OIDCTrustPolicies []OIDCTrustPolicyModel `tfsdk:"oidc_trust_policies"`
+}
+
+// Ensure provider defined types fully satisfy framework interfaces
+var (
+	_ resource.Resource                = &serviceAccountResource{}
+	_ resource.ResourceWithConfigure   = &serviceAccountResource{}
+	_ resource.ResourceWithImportState = &serviceAccountResource{}
+)
```

Comment:
Same comment as before

```suggestion
var (
	_ resource.Resource                = (*serviceAccountResource)(nil)
	_ resource.ResourceWithConfigure   = (*serviceAccountResource)(nil)
	_ resource.ResourceWithImportState = (*serviceAccountResource)(nil)
)
```